<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
<html>
<head>
<META http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="version" content="6.0.3">
<link rel="shortcut icon" href="../icons/favicon.ico" type="image/vnd.microsoft.icon">
<title>Exceptions</title>
<link href="man.css" type="text/css" rel="stylesheet">
<meta name="organisation" content="Quality First Software GmbH">
<meta name="copyright" content="(C) 1999-2022 Quality First Software GmbH">
</head>
<body bgcolor="white">
<a name="sec_exceptions"></a>
<table width="100%" border="0" cellspacing="0" cellpadding="0" class="naviheader">
<tr>
<td class="margin"></td><td class="navicon"><a href="manual.html"><img src="../icons/linktop.png" width="34" height="36" border="0" alt="Top" title="Top"></a></td><td class="navicon"><a href="contents.html#table-of-contents"><img src="../icons/linktoc.png" width="34" height="36" border="0" alt="Inhaltsverzeichnis" title="Inhaltsverzeichnis"></a></td><td class="navicon"><a href="nodes.html#sec_suite"><img src="../icons/linkprev.png" width="34" height="36" border="0" alt="Vorheriges Kapitel: 38&nbsp;Bestandteile einer Testsuite" title="Vorheriges Kapitel: 38&nbsp;Bestandteile einer Testsuite"></a></td><td class="navicon"><a href="technical.html#sec_technical"><img src="../icons/linkcont.png" width="34" height="36" border="0" alt="N&auml;chste Seite: IV&nbsp;Technische Referenz" title="N&auml;chste Seite: IV&nbsp;Technische Referenz"></a></td><td class="navicon"><img src="../icons/linknextdis.png" width="34" height="36" alt="" border="0"></td><td class="navicon"><a href="reference.html#sec_reference"><img src="../icons/linkup.png" width="34" height="36" border="0" alt="Kapitel-&Uuml;bersicht: III&nbsp;Referenzteil" title="Kapitel-&Uuml;bersicht: III&nbsp;Referenzteil"></a></td><td class="navicon"><a href="../../manual_de.pdf" target="_parent"><img src="../icons/linkpdf.png" width="34" height="36" border="0" alt="PDF Version" title="PDF Version"></a></td><td class="homeicon"><a href="http://www.qfs.de/de/qftest/index.html" target="_parent"><img src="../icons/qftest.png" width="127" height="42" border="0" alt="QF-Test" title="QF-Test"></a></td>
</tr>
</table>
<table width="100%" border="0" cellspacing="0" cellpadding="0" class="naviversion">
<tr>
<td>Version 6.0.3</td>
</tr>
</table>
    
<h3 class="header-container h3">
<div>
<span class="numtitle"><a href="contents.html#toc_sec_exceptions">39</a></span>
</div>
<a href="contents.html#toc_sec_exceptions">Exceptions</a>
</h3>
    
<p>
      Es gibt viele Exceptions, die bei der Ausf&uuml;hrung eines Tests
      mit QF-Test geworfen werden k&ouml;nnen. Dieses Kapitel f&uuml;hrt alle Exceptions
      mit ihren typischen Fehlermeldungen und einer kurzen Erkl&auml;rung auf.
    </p>
    
<p>Wenn Sie mit Exceptions in Skripten arbeiten wollen, werfen
      Sie bitte ein Blick in <a href="tech_scripting.html#sec_api_exceptions">Abschnitt 46.8</a>.
    </p>
    
<dl class="exception">
      

      
<a name="ex_TestException"></a>
<dt>
          
<a name="idx_N153254"></a><code>TestException</code>
        
</dt>
<dd>
          
<p>
            Dies ist die Basisklasse aller QF-Test Exceptions. Eine konkrete
            Exception ist nie von dieser, sondern immer von einer davon
            abgeleiteten Klasse.  Ein <a href="control.html#step_CatchSequence" shape="rect">'Catch'</a> Knoten, dessen
            <a href="control.html#att_CatchSequence_exception" shape="rect">'Exception Klasse'</a> auf <code>TestException</code> gesetzt ist,
            f&auml;ngt alle m&ouml;glichen Exceptions. Wie in Java selbst sollten Sie
            einen solchen <a href="control.html#step_CatchSequence" shape="rect">'Catch'</a> nur in Ausnahmef&auml;llen
            verwenden, da dadurch Exceptions verschleiert werden k&ouml;nnen, mit
            denen zun&auml;chst nicht gerechnet wurde.
          </p>
        
</dd>

      

      

      
<a name="ex_ComponentNotFoundException"></a>
<dt>
          
<a name="idx_N153297"></a><code>ComponentNotFoundException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die Zielkomponente eines <a href="events.html#sec_events" shape="rect">Events</a> oder <a href="checks.html#sec_checks" shape="rect">Checks</a> nicht ermittelt werden kann.  Auch
            das Scheitern eines <a href="miscnodes.html#step_ComponentWaiter" shape="rect">'Warten auf Komponente'</a> Knotens f&uuml;hrt zu
            einer <code>ComponentNotFoundException</code>, sofern dessen
            <a href="miscnodes.html#att_ComponentWaiter_absence" shape="rect">'Warten auf Verschwinden'</a> Attribut nicht gesetzt ist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Zielkomponente wurde nicht gefunden.
        </p>
</dd>

      
      

      
<a name="ex_ScopeNotFoundException"></a>
<dt>
          
<a name="idx_N153354"></a><code>ScopeNotFoundException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die Zielkomponente f&uuml;r den expliziten Scope eines <a href="events.html#sec_events" shape="rect">Events</a> oder <a href="checks.html#sec_checks" shape="rect">Checks</a> nicht ermittelt werden kann.  Auch
            das Scheitern eines <a href="miscnodes.html#step_ComponentWaiter" shape="rect">'Warten auf Komponente'</a> Knotens mit einem expliziten Scope f&uuml;hrt zu
            einer <code>ScopeNotFoundException</code>, sofern dessen
            <a href="miscnodes.html#att_ComponentWaiter_absence" shape="rect">'Warten auf Verschwinden'</a> Attribut nicht gesetzt ist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_ComponentNotFoundException" shape="rect"><code>ComponentNotFoundException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Zielkomponente f&uuml;r den expliziten Scope wurde nicht gefunden.
          Dies kann sehr unterschiedliche Ursachen haben.
          Weitere Informationen und konkrete Hilfestellung zu diesem Problem finden Sie in Kapitel 5 des Handbuchs.
        </p>
</dd>

      
      

      
<a name="ex_DocumentNotLoadedException"></a>
<dt>
          
<a name="idx_N153411"></a><code>DocumentNotLoadedException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception ist eine Variante der <code>ComponentNotFoundException</code> und wird speziell
            beim Fehlschlagen eines <a href="miscnodes.html#step_DocumentWaiter" shape="rect">'Warten auf Laden des Dokuments'</a> Knotens geworfen.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_ComponentNotFoundException" shape="rect"><code>ComponentNotFoundException</code></a>
</p>
<p>
<b>Meldung: </b>
          Das Dokument wurde nicht (neu) geladen.
        </p>
</dd>

      
      

      
<a name="ex_PageNotFoundException"></a>
<dt>
          
<a name="idx_N153452"></a><code>PageNotFoundException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception ist eine Variante der <code>ComponentNotFoundException</code> und wird speziell
            beim Fehlschlagen eines <a href="events.html#step_SelectionEventStep" shape="rect">'Auswahl'</a> Knotens geworfen, wenn in einem
            PDF-Dokument die anzuzeigende Seite nicht gefunden wurde.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_ComponentNotFoundException" shape="rect"><code>ComponentNotFoundException</code></a>
</p>
<p>
<b>Meldung: </b>
            Die gew&uuml;nschte Seite "Seitenname" wurde nicht gefunden.
        </p>
</dd>

      
      

      
<a name="ex_ComponentFoundException"></a>
<dt>
          
<a name="idx_N153493"></a><code>ComponentFoundException</code>
        
</dt>
<dd>
          
<p>
            Dies ist das Gegenst&uuml;ck zur <code>ComponentNotFoundException</code> und
            wird geworfen, wenn ein <a href="miscnodes.html#step_ComponentWaiter" shape="rect">'Warten auf Komponente'</a> mit gesetztem
            <a href="miscnodes.html#att_ComponentWaiter_absence" shape="rect">'Warten auf Verschwinden'</a> Attribut scheitert.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Zielkomponente existiert.
        </p>
</dd>

      
      

      
<a name="ex_ModalDialogException"></a>
<dt>
          
<a name="idx_N153540"></a><code>ModalDialogException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn ein <a href="events.html#sec_events" shape="rect">Event</a> von einem modalen Dialog blockiert
            wird. Siehe Option <a href="opt_play.html#opt_checkmodaldialog" shape="rect">Auf modale
            Dialoge pr&uuml;fen</a>.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Ein modaler Dialog hat diesen Event blockiert. (Plus Information
          &uuml;ber den Dialog).
        </p>
</dd>

      
      

      
<a name="ex_ComponentCannotGetFocusException"></a>
<dt>
          
<a name="idx_N153582"></a><code>ComponentCannotGetFocusException</code>
        
</dt>
<dd>
          
<p>
            
<span class="margin"><span class="note">Hinweis</span></span> Diese Ausnahme ist veraltet und sollte nicht
             mehr auftreten.
          </p>
          
<p>
            Diese Exception wird geworfen, wenn die Zielkomponente f&uuml;r einen
            <a href="events.html#step_KeyEventStep" shape="rect">'Tastaturevent'</a> oder <a href="events.html#step_TextInputStep" shape="rect">'Texteingabe'</a> Knoten ein
            Textfeld ist, das aus irgendwelchen Gr&uuml;nden den Tastaturfokus
            nicht erhalten kann. Mit JDK&nbsp;1.4 f&uuml;hrt dies dazu, dass der Event
            nicht an die Komponente &uuml;bermittelt werden kann.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Zielkomponente kann den Tastaturfokus nicht erhalten.
        </p>
</dd>

      
      

      
<a name="ex_DisabledComponentException"></a>
<dt>
          
<a name="idx_N153631"></a><code>DisabledComponentException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die Zielkomponente f&uuml;r einen
            <a href="events.html#step_MouseEventStep" shape="rect">'Mausevent'</a>, <a href="events.html#step_KeyEventStep" shape="rect">'Tastaturevent'</a> oder <a href="events.html#step_TextInputStep" shape="rect">'Texteingabe'</a> Knoten
            deaktiviert ist. In diesem Fall w&uuml;rden die Events einfach ignoriert und damit
            h&ouml;chstwahrscheinlich zu Fehlern im weiteren Verlauf des Tests f&uuml;hren.
          </p>
          
<p>
            Aus Gr&uuml;nden der R&uuml;ckw&auml;rtskompatibilit&auml;t kann diese Exception durch
            Deaktivieren der Option <a href="opt_play.html#opt_throwdisabled">DisabledComponentExceptions werfen</a>
            unterdr&uuml;ckt werden.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Zielkomponente ist deaktiviert und kann daher keine Events
          empfangen.
        </p>
</dd>

      
      

      
<a name="ex_DisabledComponentStepException"></a>
<dt>
          
<a name="idx_N153687"></a><code>DisabledComponentStepException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die Zielkomponente f&uuml;r einen
            <a href="events.html#step_MouseEventStep" shape="rect">'Mausevent'</a>, <a href="events.html#step_KeyEventStep" shape="rect">'Tastaturevent'</a> oder
            <a href="events.html#step_TextInputStep" shape="rect">'Texteingabe'</a> Knoten unter 'Fenster und Komponenten'
            deaktiviert ist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Zielkomponente ist unter 'Fenster und Komponenten' deaktiviert und kann
          daher keine Events empfangen.
        </p>
</dd>

      
      

      
<a name="ex_BadItemException"></a>
<dt>
          
<a name="idx_N153743"></a><code>BadItemException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird von Item-Resolvern geworfen, wenn der Typ des Elements
            und des Items nicht passen.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
</dd>

      
      

      
<a name="ex_ExecutionTimeoutExpiredException"></a>
<dt>
          
<a name="idx_N153772"></a><code>ExecutionTimeoutExpiredException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die angegebene maximale Ausf&uuml;hrungszeit eines Knotens
            abgelaufen ist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Zeitbeschr&auml;nkung f&uuml;r Ausf&uuml;hrung von ... ms abgelaufen.
        </p>
</dd>

      
      

      
<a name="ex_BusyPaneException"></a>
<dt>
          
<a name="idx_N153804"></a><code>BusyPaneException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die Zielkomponente f&uuml;r einen
            <a href="events.html#step_MouseEventStep" shape="rect">'Mausevent'</a>, <a href="events.html#step_KeyEventStep" shape="rect">'Tastaturevent'</a> oder <a href="events.html#step_TextInputStep" shape="rect">'Texteingabe'</a> Knoten von
            einer GlassPane mit einem 'busy' Mauszeiger verdeckt ist. In diesem Fall w&uuml;rden die
            Events normalerweise von der GlassPane abgefangen werden und damit
            h&ouml;chstwahrscheinlich zu Fehlern im weiteren Verlauf des Tests f&uuml;hren. Die Option
            <a href="opt_play.html#opt_waitforbusy">
              Warten bei 'busy' GlassPane (ms)</a> legt fest, wie lange QF-Test auf das Verschwinden der
            GlassPane wartet, bevor die Exception geworfen wird.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Zielkomponente wird von einer 'busy' GlassPane verdeckt.
        </p>
</dd>

      
      

      
<a name="ex_InvisibleDnDTargetException"></a>
<dt>
          
<a name="idx_N153857"></a><code>InvisibleDnDTargetException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn der Zielpunkt in der Komponente f&uuml;r einen
            <a href="events.html#step_MouseEventStep" shape="rect">'Mausevent'</a> vom Typ <code>DRAG_FROM</code>, <code>DRAG_OVER</code> oder
            <code>DROP_TO</code> f&uuml;r eine Drag&amp;Drop Operation nicht sichtbar ist und nicht durch
            Scrollen der Zielkomponente sichtbar gemacht werden kann.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Zielpunkt der Drag&amp;Drop Operation konnte nicht sichtbar gemacht werden.
        </p>
</dd>

      
      

      
<a name="ex_InvisibleTargetComponentException"></a>
<dt>
          
<a name="idx_N153904"></a><code>InvisibleTargetComponentException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die Zielkomponente unsichtbar ist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_ComponentNotFoundException" shape="rect"><code>ComponentNotFoundException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Zielkomponente ist unsichtbar.
        </p>
</dd>

      
      

      
<a name="ex_InvisibleTargetItemException"></a>
<dt>
          
<a name="idx_N153936"></a><code>InvisibleTargetItemException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn zwar die umgebende Zielkomponente sichtbar ist, aber das Zielelement darin unsichtbar.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_IndexNotFoundException" shape="rect"><code>IndexNotFoundException</code></a>
</p>
<p>
<b>Meldung: </b>
          Das Zielelement mit dem Index '...' ist unsichtbar.
        </p>
</dd>

      
      

      
<a name="ex_DeadlockTimeoutException"></a>
<dt>
          
<a name="idx_N153968"></a><code>DeadlockTimeoutException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn das SUT f&uuml;r einen
            vorgegebenen Zeitraum nicht reagiert. Dieser Zeitraum wird durch
            die Option <a href="opt_play.html#opt_deadlocktimeout">Erkennen von Deadlocks (s)</a> festgelegt.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Das Zeitlimit f&uuml;r die Operation wurde &uuml;berschritten.
          Es ist sehr wahrscheinlich, dass der Client
          '...' sich in einem Deadlock befindet.
        </p>
</dd>

      
      

      
<a name="ex_DownloadNotCompleteException"></a>
<dt>
          
<a name="idx_N154003"></a><code>DownloadNotCompleteException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn auf das Herunterladen eines Dokuments gewartet
            wird, dies aber nicht abgeschlossen wurde.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_ComponentNotFoundException" shape="rect"><code>ComponentNotFoundException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Download der Datei '...' ist noch nicht abgeschlossen.
        </p>
</dd>

      
      

      
<a name="ex_DownloadStillActiveException"></a>
<dt>
          
<a name="idx_N154035"></a><code>DownloadStillActiveException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn versucht wird, eine Datei herunter
            zu laden, die noch durch einen anderen Download blockiert ist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_ComponentNotFoundException" shape="rect"><code>ComponentNotFoundException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Download der Datei '...' ist noch aktiv, so dass Sie keinen weiteren
          Download f&uuml;r diese Datei starten k&ouml;nnen.
        </p>
</dd>

      
      

      
<a name="ex_NoSuchDownloadException"></a>
<dt>
          
<a name="idx_N154067"></a><code>NoSuchDownloadException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn auf einen Download gewartet wird,
            bei dem das Herunterladen nicht gestartet hat.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_ComponentNotFoundException" shape="rect"><code>ComponentNotFoundException</code></a>
</p>
<p>
<b>Meldung: </b>
          Es ist noch kein Download f&uuml;r die Datei '...' gestartet worden.
        </p>
</dd>

      

      

      
<a name="ex_VariableException"></a>
<dt>
          
<a name="idx_N154099"></a><code>VariableException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird nicht selbst geworfen, sondern ist die
            Basisklasse f&uuml;r verschiedene Exceptions, die im Zusammenhang mit
            Variablenexpansion geworfen werden k&ouml;nnen.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
</dd>

      
      

      
<a name="ex_BadVariableSyntaxException"></a>
<dt>
          
<a name="idx_N154128"></a><code>BadVariableSyntaxException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn ein Ausdruck bei der
            Variablenexpansion keine g&uuml;ltige Syntax aufweist, z.B. weil die
            schlie&szlig;ende Klammer fehlt.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_VariableException" shape="rect"><code>VariableException</code></a>
</p>
<p>
<b>Meldung: </b>
          Ung&uuml;ltige Variablensyntax: '...'
        </p>
</dd>

      
      

      
<a name="ex_MissingPropertiesException"></a>
<dt>
          
<a name="idx_N154160"></a><code>MissingPropertiesException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn keine Properties oder kein
            ResourceBundle f&uuml;r die Gruppe einer erweiterten Variablenexpansion
            der Form <code>${<em>Gruppe</em>:<em>Name</em>}</code> verf&uuml;gbar
            sind (siehe auch <a href="miscnodes.html#step_LoadResourcesStep" shape="rect">'Ressourcen laden'</a> und
            <a href="miscnodes.html#step_LoadPropertiesStep" shape="rect">'Properties laden'</a>).
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_VariableException" shape="rect"><code>VariableException</code></a>
</p>
<p>
<b>Meldung: </b>
          Unter dem Namen '...' waren keine Properties oder Ressourcen
          verf&uuml;gbar.
        </p>
</dd>

      
      

      
<a name="ex_MissingPropertyException"></a>
<dt>
          
<a name="idx_N154213"></a><code>MissingPropertyException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn der Name bei einer erweiterten
            Variablenexpansion der Form
            <code>${<em>Gruppe</em>:<em>Name</em>}</code> nicht verf&uuml;gbar ist
            (siehe auch <a href="miscnodes.html#step_LoadResourcesStep" shape="rect">'Ressourcen laden'</a> und
            <a href="miscnodes.html#step_LoadPropertiesStep" shape="rect">'Properties laden'</a>).
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_VariableException" shape="rect"><code>VariableException</code></a>
</p>
<p>
<b>Meldung: </b>
          Es gibt keine Property oder Ressource namens '...' in '...'.
        </p>
</dd>

      
      

      
<a name="ex_RecursiveVariableException"></a>
<dt>
          
<a name="idx_N154266"></a><code>RecursiveVariableException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die Expansion einer Variablen
            zu Rekursion f&uuml;hrt, z.B. wenn Sie eine Variable namens
            <code>x</code> auf den Wert <code>$(y)</code> und die Variable
            namens <code>y</code> auf <code>$(x)</code> setzen und versuchen,
            den Wert <code>$(x)</code> zu expandieren.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_VariableException" shape="rect"><code>VariableException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Wert der Variable '...' konnte aufgrund rekursiver
          Zuweisung nicht ausgewertet werden.
        </p>
</dd>

      
      

      
<a name="ex_UnboundVariableException"></a>
<dt>
          
<a name="idx_N154313"></a><code>UnboundVariableException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn eine Variable bei einer
            Variablenexpansion nicht existiert.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_VariableException" shape="rect"><code>VariableException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Variable '...' ist unbekannt.
        </p>
</dd>

      
      

      
<a name="ex_VariableNumberException"></a>
<dt>
          
<a name="idx_N154345"></a><code>VariableNumberException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn das Ergebnis einer
            Variablenexpansion f&uuml;r ein numerisches Attribut keine Zahl
            darstellt.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_VariableException" shape="rect"><code>VariableException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Wert '...' ist keine g&uuml;ltige Zahl.
        </p>
</dd>

      

      

      
<a name="ex_BadExpressionException"></a>
<dt>
          
<a name="idx_N154377"></a><code>BadExpressionException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn ein Ausdruck der Form $[...]
            fehlerhaft ist (vgl. <a href="user_variables.html#usec_expressions">Abschnitt 6.6</a>).
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_ScriptException" shape="rect"><code>ScriptException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Ausdruck '...' ist ung&uuml;ltig.
        </p>
</dd>

      
      

      
<a name="ex_BadTestException"></a>
<dt>
          
<a name="idx_N154412"></a><code>BadTestException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die <a href="control.html#att_IfSequence_test" shape="rect">'Bedingung'</a> eines
            <a href="control.html#step_IfSequence" shape="rect">'If'</a> oder <a href="control.html#step_ElseifSequence" shape="rect">'Elseif'</a> Knotens eine
            ung&uuml;ltige Syntax aufweist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_ScriptException" shape="rect"><code>ScriptException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Ausdruck '...' ist keine g&uuml;ltige Bedingung.
        </p>
</dd>

      
      

      
<a name="ex_BadRegexpException"></a>
<dt>
          
<a name="idx_N154462"></a><code>BadRegexpException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn ein regul&auml;rer Ausdruck keine
            g&uuml;ltige Syntax aufweist (vgl <a href="tech_regexp.html#sec_regexp">Abschnitt 45.3</a>), z.B.
            bei einem <a href="components.html#step_SubItemStep" shape="rect">'Element'</a> oder <a href="checks.html#step_CheckStringStep" shape="rect">'Check Text'</a>
            Knoten.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Wert '...' ist kein g&uuml;ltiger regul&auml;rer Ausdruck.
        </p>
</dd>

      
      

      
<a name="ex_BadRangeException"></a>
<dt>
          
<a name="idx_N154509"></a><code>BadRangeException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn das Attribut 'Iterationsbereiche' eines 'Daten'
            Knotens ung&uuml;ltige Syntax aufweist oder einen Index au&szlig;erhalb des Datenbereichs enth&auml;lt.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Ung&uuml;ltige Syntax f&uuml;r Iterationsbereiche: '...'.<br>
          Ung&uuml;ltiger Index im Iterationsbereich: '...'.
        </p>
</dd>

      

      

      
<a name="ex_CannotExecuteException"></a>
<dt>
          
<a name="idx_N154549"></a><code>CannotExecuteException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn der Start eines Prozesses
            durch einen <a href="processes.html#step_JavaClientStarter" shape="rect">'Java SUT Client starten'</a> Knoten fehlschl&auml;gt.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Fehler beim Starten des Prozesses '...'. Bitte &uuml;berpr&uuml;fen Sie die
          Angaben und die PATH Einstellung.
        </p>
</dd>

      
      

      
<a name="ex_InvalidDirectoryException"></a>
<dt>
          
<a name="idx_N154587"></a><code>InvalidDirectoryException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn das
            <a href="processes.html#att_JavaClientStarter_directory" shape="rect">'Verzeichnis'</a> Attribut eines
            <a href="processes.html#step_JavaClientStarter" shape="rect">'Java SUT Client starten'</a> Knotens auf ein nicht existierendes
            Verzeichnis verweist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Das Verzeichnis '...' existiert nicht.
        </p>
</dd>

      

      

      
<a name="ex_CheckFailedException"></a>
<dt>
          
<a name="idx_N154631"></a><code>CheckFailedException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn ein <a href="checks.html#sec_checks" shape="rect">Check</a> fehlschl&auml;gt, dessen
            <a href="checks.html#att_CheckStringStep_raise" shape="rect">'Im Fehlerfall Exception werfen'</a> Attribut gesetzt ist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Check f&uuml;r Client '...' ist fehlgeschlagen. Erwartet:
          '...'. Erhalten: '...'.
        </p>
</dd>

      
      

      
<a name="ex_CheckNotSupportedException"></a>
<dt>
          
<a name="idx_N154674"></a><code>CheckNotSupportedException</code>
        
</dt>
<dd>
          
<p>
            Wie im Abschnitt &uuml;ber <a href="checks.html#sec_checks" shape="rect">Checks</a>
            beschrieben, kann sich jeder Check nur auf bestimmte Komponenten
            beziehen. Diese Exception wird geworfen, wenn
            die Zielkomponente eines Checks f&uuml;r diesen keinen Sinn ergibt.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Dieser Check ist f&uuml;r eine Komponente der Klasse '...' nicht m&ouml;glich.
        </p>
</dd>

      
      

      
<a name="ex_OperationNotSupportedException"></a>
<dt>
          
<a name="idx_N154711"></a><code>OperationNotSupportedException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn eine Operation wie
            <a href="queries.html#step_FetchTextStep" shape="rect">'Text auslesen'</a> f&uuml;r die angegebene Zielkomponente nicht
            m&ouml;glich ist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Diese Operation ist f&uuml;r eine Komponente der Klasse '...' nicht
          m&ouml;glich.
        </p>
</dd>

      
      

      
<a name="ex_BadComponentException"></a>
<dt>
          
<a name="idx_N154749"></a><code>BadComponentException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn eine Komponente f&uuml;r einen <a href="events.html#sec_events" shape="rect">Event</a> keinen Sinn ergibt, z.B. etwas
            anderes als ein Fenster f&uuml;r einen <a href="events.html#step_WindowEventStep" shape="rect">'Fensterevent'</a>.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Komponente der Klasse '...' ist kein Fenster.
        </p>
</dd>

      

      

      
<a name="ex_IndexFormatException"></a>
<dt>
          
<a name="idx_N154792"></a><code>IndexFormatException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn der Index eines Unterelements
            in einem ung&uuml;ltigen Format angegeben ist (vgl. <a href="user_subitems.html#sec_directsubitemeasy">Abschnitt 5.9.1</a>).
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Wert '...' ist kein g&uuml;ltiger Textindex/Zahl/regul&auml;rer Ausdruck.
        </p>
</dd>

      
      

      
<a name="ex_IndexFoundException"></a>
<dt>
          
<a name="idx_N154827"></a><code>IndexFoundException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn ein Unterelement gefunden wird, obwohl bei der
            Ausf&uuml;hrung eines <a href="miscnodes.html#step_ComponentWaiter" shape="rect">'Warten auf Komponente'</a> Knotens auf dessen Abwesenheit gepr&uuml;ft wird.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_ComponentFoundException" shape="rect"><code>ComponentFoundException</code></a>
</p>
<p>
<b>Meldung: </b>
          Kein Element f&uuml;r Index '...' gefunden.
        </p>
</dd>

      
      

      
<a name="ex_IndexNotFoundException"></a>
<dt>
          
<a name="idx_N154865"></a><code>IndexNotFoundException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn kein Unterelement f&uuml;r einen
            angegebenen Index gefunden wurde.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_ComponentNotFoundException" shape="rect"><code>ComponentNotFoundException</code></a>
</p>
<p>
<b>Meldung: </b>
          Kein Element f&uuml;r Index '...' gefunden.
        </p>
</dd>

      
      

      
<a name="ex_IndexRequiredException"></a>
<dt>
          
<a name="idx_N154897"></a><code>IndexRequiredException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn ein f&uuml;r eine Operation
            notwendiger Index nicht angegeben wurden, z.B. f&uuml;r einen
            <a href="checks.html#step_CheckStringStep" shape="rect">'Check Text'</a> Knoten f&uuml;r einen <code>JTree</code>.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Diese Operation erfordert einen Index.
        </p>
</dd>

      
      

      
<a name="ex_UnexpectedIndexException"></a>
<dt>
          
<a name="idx_N154938"></a><code>UnexpectedIndexException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn ein Index f&uuml;r ein Unterelement
            angegeben ist, obwohl eine Operation keinen ben&ouml;tigt, z.B.  f&uuml;r
            einen <a href="checks.html#step_CheckItemsStep" shape="rect">'Check Elemente'</a> Knoten f&uuml;r einen <code>JTree</code>.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Diese Operation unterst&uuml;tzt f&uuml;r eine Komponente der Klasse '...'
          keinen Index.
        </p>
</dd>

      

      

      
<a name="ex_ClientNotConnectedException"></a>
<dt>
          
<a name="idx_N154979"></a><code>ClientNotConnectedException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn keine Verbindung zum
            angegebenen SUT Client f&uuml;r eine Operation besteht, obwohl
            im Gegensatz zur <a href="#ex_NoSuchClientException" shape="rect"><code>NoSuchClientException</code></a> ein Prozess unter
            diesem Namen verf&uuml;gbar ist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Es besteht keine Verbindung mit Client '...'.
        </p>
</dd>

      
      

      
<a name="ex_CannotAttachException"></a>
<dt>
          
<a name="idx_N155017"></a><code>CannotAttachException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, die Verbindung zu der Windows Anwendung fehlschlug.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Fehler beim Verbinden mit der Anwendung '... Titel: ...'.
        </p>
</dd>

      
      

      
<a name="ex_ConnectionFailureException"></a>
<dt>
          
<a name="idx_N155049"></a><code>ConnectionFailureException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die Verbindung zu einem Client fehlerhaft ist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Verbindung zu Client '...' ist fehlerhaft.
        </p>
</dd>

      
      

      
<a name="ex_NoSuchClientException"></a>
<dt>
          
<a name="idx_N155081"></a><code>NoSuchClientException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn der angegebene SUT Client
            f&uuml;r eine Operation nicht existiert.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Client '...' existiert nicht.
        </p>
</dd>

      
      

      
<a name="ex_NoSuchEngineException"></a>
<dt>
          
<a name="idx_N155113"></a><code>NoSuchEngineException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn eine Engine referenziert wird,
            die nicht existiert oder wenn versucht wird, auf eine Engine zur&uuml;ckzugreifen,
            die (noch) nicht mit QF-Test verbunden wurde Dies kann passieren, wenn die
            entsprechende Technologie in der Anwendung nicht oder erst zu einem sp&auml;teren
            Zeitpunkt verwendet wird.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Es gibt keine Engine namens '...' im Client '...'.
        </p>
</dd>

      
      

      
<a name="ex_DuplicateClientException"></a>
<dt>
          
<a name="idx_N155145"></a><code>DuplicateClientException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn versucht wird, mehr als einen
            Prozess gleichzeitig unter dem selben Namen zu starten.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Es wurde bereits ein Client namens '...' gestartet.
        </p>
</dd>

      
      

      
<a name="ex_UnexpectedClientException"></a>
<dt>
          
<a name="idx_N155177"></a><code>UnexpectedClientException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn im SUT eine unerwartete
            Exception beim Abspielen eines Events geworfen wird. Sofern es
            sich dabei nicht um einen Bug in QF-Test handelt, deutet dies auf
            ein echtes Problem im SUT hin.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Unerwartete Exception in Client '...'.
        </p>
</dd>

      
      

      
<a name="ex_ExtensionException"></a>
<dt>
          
<a name="idx_N155209"></a><code>ExtensionException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn in einem Client eine unerwartete
            Exception auftritt.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Exception in externem Code: ...
        </p>
</dd>

      
      

      
<a name="ex_ClientNotTerminatedException"></a>
<dt>
          
<a name="idx_N155241"></a><code>ClientNotTerminatedException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn bei der Ausf&uuml;hrung eines
            <a href="processes.html#step_ProcessWaiter" shape="rect">'Warten auf Programmende'</a> Knotens der Prozess nicht terminiert.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Prozess '...' wurde nicht beendet.
        </p>
</dd>

      
      

      
<a name="ex_UnexpectedExitCodeException"></a>
<dt>
          
<a name="idx_N155279"></a><code>UnexpectedExitCodeException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn der Exitcode eines Prozesses
            nicht den Vorgaben des Attributs <a href="processes.html#att_ProcessWaiter_check" shape="rect">'Erwarteter Exitcode'</a>
            eines <a href="processes.html#step_ProcessWaiter" shape="rect">'Warten auf Programmende'</a> Knotens entspricht.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Falscher Exitcode.
        </p>
</dd>

      
      

      
<a name="ex_BadExitCodeException"></a>
<dt>
          
<a name="idx_N155323"></a><code>BadExitCodeException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn das Attribut
            <a href="processes.html#att_ProcessWaiter_check" shape="rect">'Erwarteter Exitcode'</a> eines <a href="processes.html#step_ProcessWaiter" shape="rect">'Warten auf Programmende'</a> Knotens
            nicht der Spezifikation entspricht.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Ung&uuml;ltiger Exitcode '...'. Bitte verwenden Sie '==', '!=', '&lt;'
          oder '&gt;' gefolgt von einer Zahl.
        </p>
</dd>

      

      

      
<a name="ex_ComponentIdMismatchException"></a>
<dt>
          
<a name="idx_N155367"></a><code>ComponentIdMismatchException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn das
            <a href="events.html#att_MouseEventStep_component" shape="rect">'QF-Test ID der Komponente'</a> Attribut eines Knotens auf einen
            Knoten verweist, der nicht vom Typ <a href="components.html#step_WindowStep" shape="rect">'Fenster'</a>,
            <a href="components.html#step_ComponentStep" shape="rect">'Komponente'</a> oder <a href="components.html#step_SubItemStep" shape="rect">'Element'</a> ist.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Knoten '...' mit der QF-Test ID '...' ist keine Komponente.
        </p>
</dd>

      
      

      
<a name="ex_UnresolvedComponentIdException"></a>
<dt>
          
<a name="idx_N155423"></a><code>UnresolvedComponentIdException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn das Ziel eines
            <a href="events.html#att_MouseEventStep_component" shape="rect">'QF-Test ID der Komponente'</a> Attributs nicht existiert.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Es gibt keine Komponente mit der QF-Test ID '...'.
        </p>
</dd>

      

      

      
<a name="ex_TestNotFoundException"></a>
<dt>
          
<a name="idx_N155461"></a><code>TestNotFoundException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn der <a href="sequences.html#step_TestCase" shape="rect">'Testfall'</a> oder das <a href="sequences.html#step_TestSet" shape="rect">'Testfallsatz'</a> f&uuml;r
            einen <a href="sequences.html#step_TestCall" shape="rect">'Testaufruf'</a> nicht existiert.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Der Test '...' ist nicht vorhanden.
        </p>
</dd>

      
      

      
<a name="ex_DependencyNotFoundException"></a>
<dt>
          
<a name="idx_N155511"></a><code>DependencyNotFoundException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die <a href="dependencies.html#step_Dependency" shape="rect">'Abh&auml;ngigkeit'</a> f&uuml;r
            einen <a href="dependencies.html#step_DependencyReference" shape="rect">'Bezug auf Abh&auml;ngigkeit'</a> nicht existiert.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Abh&auml;ngigkeit '...' ist nicht vorhanden.
        </p>
</dd>

      
      

      
<a name="ex_InconsistentDependenciesException"></a>
<dt>
          
<a name="idx_N155555"></a><code>InconsistentDependenciesException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die <a href="dependencies.html#step_Dependency" shape="rect">'Abh&auml;ngigkeit'</a>
            aufgrund inkonsistenter Bez&uuml;ge nicht linearisiert werden k&ouml;nnen.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die <a href="dependencies.html#step_Dependency" shape="rect">'Abh&auml;ngigkeit'</a> k&ouml;nnen aufgrund inkonsistenter Bez&uuml;ge nicht linearisiert werden.
        </p>
</dd>

      
      

      
<a name="ex_RecursiveDependencyReferenceException"></a>
<dt>
          
<a name="idx_N155599"></a><code>RecursiveDependencyReferenceException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die <a href="dependencies.html#step_Dependency" shape="rect">'Abh&auml;ngigkeit'</a>
            aufgrund rekursiver Bez&uuml;ge nicht linearisiert werden k&ouml;nnen.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die <a href="dependencies.html#step_Dependency" shape="rect">'Abh&auml;ngigkeit'</a> k&ouml;nnen aufgrund rekursiver Bez&uuml;ge nicht linearisiert werden.
        </p>
</dd>

      

      

      
<a name="ex_ProcedureNotFoundException"></a>
<dt>
          
<a name="idx_N155643"></a><code>ProcedureNotFoundException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die <a href="procedures.html#step_Procedure" shape="rect">'Prozedur'</a> f&uuml;r
            einen <a href="procedures.html#step_ProcedureCall" shape="rect">'Prozeduraufruf'</a> nicht existiert.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die Prozedur '...' ist nicht vorhanden.
        </p>
</dd>

      
      

      
<a name="ex_StackOverflowException"></a>
<dt>
          
<a name="idx_N155687"></a><code>StackOverflowException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn die Verschachtelungstiefe von
            <a href="procedures.html#step_ProcedureCall" shape="rect">'Prozeduraufrufen'</a> den Wert der Option <a href="opt_play.html#opt_callstacksize" shape="rect">Gr&ouml;&szlig;e des Callstacks</a> &uuml;berschreitet,
            was auf eine endlose Rekursion von Prozeduraufrufen hindeutet.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Zu viele verschachtelte Prozeduraufrufe.
        </p>
</dd>

      

      

      
<a name="ex_UserException"></a>
<dt>
          
<a name="idx_N155730"></a><code>UserException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird explizit durch einen <a href="control.html#step_ThrowStep" shape="rect">'Throw'</a>
            Knoten geworfen.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Die <a href="control.html#att_ThrowStep_throwmessage" shape="rect">'Meldung f&uuml;r die Exception'</a> des <a href="control.html#step_ThrowStep" shape="rect">'Throw'</a> Knotens.
        </p>
</dd>

      
      

      
<a name="ex_CannotRethrowException"></a>
<dt>
          
<a name="idx_N155780"></a><code>CannotRethrowException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn versucht wird, durch einen
            <a href="control.html#step_RethrowStep" shape="rect">'Rethrow'</a> Knoten eine Exception erneut zu werfen, ohne
            dass vorher eine Exception durch einen <a href="control.html#step_CatchSequence" shape="rect">'Catch'</a>
            Knoten abgefangen wurde.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Erneutes werfen einer abgefangenen Exception ist nur innerhalb eines
          Catch Knotens m&ouml;glich.
        </p>
</dd>

      
      

      
<a name="ex_ScriptException"></a>
<dt>
          
<a name="idx_N155824"></a><code>ScriptException</code>
        
</dt>
<dd>
          
<p>
            Diese Exception wird geworfen, wenn ein Script einen Fehler
            liefert.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Das Skript ist mit der folgenden Meldung fehlgeschlagen:...
        </p>
</dd>

      

      

      
        
<a name="ex_AndroidSdkException"></a>
<dt>
            
<a name="idx_N155856"></a><code>AndroidSdkException</code>
          
</dt>
<dd>
            
<p>
              Dies ist eine Basisklasse. Um Android Anwendungen zu testen,
              ben&ouml;tigt QF-Test ein auf dem Computer installiertes Android SDK.
              Sollte es hierbei zu Problemen kommen, so wird eine Exception die
              von dieser Basisklasse abgeleitet ist geworfen.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
</dd>
      

      
      

      
        
<a name="ex_InvalidAndroidSdkPathException"></a>
<dt>
            
<a name="idx_N155885"></a><code>InvalidAndroidSdkPathException</code>
          
</dt>
<dd>
            
<p>
              Eine Exceptions dieses Typs wird geworfen, wenn ein angegeber Pfad
              nicht auf ein g&uuml;ltiges SDK verweist.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_AndroidSdkException" shape="rect"><code>AndroidSdkException</code></a>
</p>
</dd>
      

      
      

      
        
<a name="ex_AndroidSdkNotFoundException"></a>
<dt>
            
<a name="idx_N155914"></a><code>AndroidSdkNotFoundException</code>
          
</dt>
<dd>
            
<p>
              Eine Exceptions dieses Typs wird geworfen wenn das Android SDK nicht auf
              dem Computer gefunden werden konnte.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_AndroidSdkException" shape="rect"><code>AndroidSdkException</code></a>
</p>
</dd>
      

      
      

      
        
<a name="ex_InvalidAndroidAdbPathException"></a>
<dt>
            
<a name="idx_N155943"></a><code>InvalidAndroidAdbPathException</code>
          
</dt>
<dd>
            
<p>
              Diese Exception wird geworfen, wenn der angegebenen ADB Klassenpfad ung&uuml;ltig ist.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_AndroidSdkException" shape="rect"><code>AndroidSdkException</code></a>
</p>
<p>
<b>Meldung: </b>
            ... ist kein valider Android ADB Pfad. (Quelle: ..., Grund: ...)
          </p>
</dd>
      

      
      

      
        
<a name="ex_AndroidAdbNotFoundException"></a>
<dt>
            
<a name="idx_N155975"></a><code>AndroidAdbNotFoundException</code>
          
</dt>
<dd>
            
<p>
              Diese Exception wird geworfen, wenn kein Android ADB auf dem Rechner gefunden wurde.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_AndroidSdkException" shape="rect"><code>AndroidSdkException</code></a>
</p>
<p>
<b>Meldung: </b>
            Es wurde kein Android ADB gefunden.
          </p>
</dd>
      

      
      

      
        
<a name="ex_InvalidAndroidEmulatorPathException"></a>
<dt>
            
<a name="idx_N156007"></a><code>InvalidAndroidEmulatorPathException</code>
          
</dt>
<dd>
            
<p>
              Diese Exception wird geworfen, wenn der angegebenen Pfad f&uuml;r den Android Emulator ung&uuml;ltig ist.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_AndroidSdkException" shape="rect"><code>AndroidSdkException</code></a>
</p>
<p>
<b>Meldung: </b>
            ... ist kein valider Android Emulator Pfad. (Quelle: ..., Grund: ...)
          </p>
</dd>
      

      
      

      
        
<a name="ex_AndroidEmulatorNotFoundException"></a>
<dt>
            
<a name="idx_N156039"></a><code>AndroidEmulatorNotFoundException</code>
          
</dt>
<dd>
            
<p>
              Diese Exception wird geworfen, wenn kein Android Emulator auf dem Rechner gefunden wurde.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_AndroidSdkException" shape="rect"><code>AndroidSdkException</code></a>
</p>
<p>
<b>Meldung: </b>
            Es wurde kein Android Emulator gefunden.
          </p>
</dd>
      

      
      

      
        
<a name="ex_AndroidVirtualDeviceException"></a>
<dt>
            
<a name="idx_N156071"></a><code>AndroidVirtualDeviceException</code>
          
</dt>
<dd>
            
<p>
              Dies ist eine Basisklasse. Um Android Anwendungen zu Testen sollte der
              Android Emulator installiert sein. Dieser Emulator kann dann ein virtuelles
              Android Ger&auml;t (AVD) ausf&uuml;hren. Wird kein solches virtuelles Android Ger&auml;t
              gefunden, oder wenn das Testen des angegebenen Emulators nicht m&ouml;glich ist,
              so wird eine Exception die von dieser Basisklasse abgeleitet ist geworfen.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
</dd>
      

      
      

      
        
<a name="ex_NoAndroidVirtualDeviceException"></a>
<dt>
            
<a name="idx_N156100"></a><code>NoAndroidVirtualDeviceException</code>
          
</dt>
<dd>
            
<p>
              Eine Exceptions dieses Typs wird geworfen wenn kein virtuelles Android Ger&auml;t
              gefunden wird auf dem die Tests ausgef&uuml;hrt werden k&ouml;nnen.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_AndroidVirtualDeviceException" shape="rect"><code>AndroidVirtualDeviceException</code></a>
</p>
</dd>
      

      
      

      
        
<a name="ex_NoSupportedAndroidVirtualDeviceException"></a>
<dt>
            
<a name="idx_N156129"></a><code>NoSupportedAndroidVirtualDeviceException</code>
          
</dt>
<dd>
            
<p>
              Eine Exceptions dieses Typs wird geworfen wenn kein von QF-Test unterst&uuml;tztes
              virtuelles Android Ger&auml;t gefunden wird auf dem die Tests ausgef&uuml;hrt werden k&ouml;nnen.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_AndroidVirtualDeviceException" shape="rect"><code>AndroidVirtualDeviceException</code></a>
</p>
</dd>
      

      
      

      
        
<a name="ex_AndroidVirtualDeviceNotFoundException"></a>
<dt>
            
<a name="idx_N156158"></a><code>AndroidVirtualDeviceNotFoundException</code>
          
</dt>
<dd>
            
<p>
              Eine Exceptions dieses Typs wird geworfen wenn das angegebene
              virtuelle Android Ger&auml;t nicht gefunden werden konnte.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_AndroidVirtualDeviceException" shape="rect"><code>AndroidVirtualDeviceException</code></a>
</p>
</dd>
      

      
      

      
        
<a name="ex_AndroidVirtualDeviceNotSupportedException"></a>
<dt>
            
<a name="idx_N156187"></a><code>AndroidVirtualDeviceNotSupportedException</code>
          
</dt>
<dd>
            
<p>
              Eine Exceptions dieses Typs wird geworfen wenn das angegebene
              virtuelle Android Ger&auml;t nicht unterst&uuml;tzt wird.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_AndroidVirtualDeviceException" shape="rect"><code>AndroidVirtualDeviceException</code></a>
</p>
</dd>
      

      
      

      
        
<a name="ex_AndroidVirtualDeviceParsingException"></a>
<dt>
            
<a name="idx_N156216"></a><code>AndroidVirtualDeviceParsingException</code>
          
</dt>
<dd>
            
<p>
              Eine Exceptions dieses Typs wird geworfen wenn es f&uuml;r QF-Test nicht m&ouml;glich ist
              die Zeichenkette zu parsen die den zu testenden Emulator spezifiziert.
            </p>
          
<p>
<b>Basisklasse: </b><a href="#ex_AndroidVirtualDeviceException" shape="rect"><code>AndroidVirtualDeviceException</code></a>
</p>
</dd>
      

      

      

      
<a name="ex_BreakException"></a>
<dt>
          
<a name="idx_N156245"></a><code>BreakException</code>
        
</dt>
<dd>
          
<p>
            Dies ist keine normale TestException und sie kann nicht von einem
            <a href="control.html#step_CatchSequence" shape="rect">'Catch'</a> Knoten gefangen werden. Sie wird von einem <a href="control.html#step_BreakStep" shape="rect">'Break'</a>
            Knoten geworfen um aus einer Schleife auszubrechen. In einem Skript erf&uuml;llt das
            Werfen einer <code>BreakException</code> den selben Zweck. Wird sie au&szlig;erhalb einer Schleife
            geworfen, l&ouml;st eine <code>BreakException</code> einen Fehler aus.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Es wurde ein Break au&szlig;erhalb einer Schleife ausgel&ouml;st.
        </p>
</dd>

      
      

      
<a name="ex_ReturnException"></a>
<dt>
          
<a name="idx_N156295"></a><code>ReturnException</code>
        
</dt>
<dd>
          
<p>
            Dies ist keine normale TestException und sie kann nicht von einem
            <a href="control.html#step_CatchSequence" shape="rect">'Catch'</a> Knoten gefangen werden. Sie wird von einem <a href="procedures.html#step_ReturnStep" shape="rect">'Return'</a>
            Knoten geworfen um aus einer <a href="procedures.html#step_Procedure" shape="rect">'Prozedur'</a> zur&uuml;ckzukehren. In einem Skript
            erf&uuml;llt das Werfen einer <code>ReturnException</code> den selben Zweck. Wird sie au&szlig;erhalb
            einer 'Prozedur' geworfen, l&ouml;st eine <code>ReturnException</code> einen Fehler aus.
          </p>
        
<p>
<b>Basisklasse: </b><a href="#ex_TestException" shape="rect"><code>TestException</code></a>
</p>
<p>
<b>Meldung: </b>
          Es wurde ein Return au&szlig;erhalb einer Prozedur ausgel&ouml;st.
        </p>
</dd>

      

      

      
<a name="ex_TestOutOfMemoryException"></a>
<dt>
          
<a name="idx_N156354"></a><code>TestOutOfMemoryException</code>
        
</dt>
<dd>
          
<p>
            Dies ist eine spezielle Exception, die geworfen wird, wenn QF-Test feststellt, dass der
            f&uuml;r die Testausf&uuml;hrung zur Verf&uuml;gung stehende Speicher zu Ende geht. Die Exception
            f&uuml;hrt zum sofortigen Abbruch des Testlaufs und kann nicht gefangen werden, da QF-Test
            ohne Speicher nicht viel tun kann, um sie zu behandeln. Allerdings versucht QF-Test, sich
            eine kleine Speicherreserve zu erhalten und zumindest das Protokoll zu speichern.
          </p>
        
</dd>

      

    
</dl>
  
<table width="100%" border="0" cellspacing="0" cellpadding="0" class="navifooter">
<tr>
<td class="margin"></td><td class="navicon"><a href="manual.html"><img src="../icons/linktop.png" width="34" height="36" border="0" alt="Top" title="Top"></a></td><td class="navicon"><a href="contents.html#table-of-contents"><img src="../icons/linktoc.png" width="34" height="36" border="0" alt="Inhaltsverzeichnis" title="Inhaltsverzeichnis"></a></td><td class="navicon"><a href="nodes.html#sec_suite"><img src="../icons/linkprev.png" width="34" height="36" border="0" alt="Vorheriges Kapitel: 38&nbsp;Bestandteile einer Testsuite" title="Vorheriges Kapitel: 38&nbsp;Bestandteile einer Testsuite"></a></td><td class="navicon"><a href="technical.html#sec_technical"><img src="../icons/linkcont.png" width="34" height="36" border="0" alt="N&auml;chste Seite: IV&nbsp;Technische Referenz" title="N&auml;chste Seite: IV&nbsp;Technische Referenz"></a></td><td class="navicon"><img src="../icons/linknextdis.png" width="34" height="36" alt="" border="0"></td><td class="navicon"><a href="reference.html#sec_reference"><img src="../icons/linkup.png" width="34" height="36" border="0" alt="Kapitel-&Uuml;bersicht: III&nbsp;Referenzteil" title="Kapitel-&Uuml;bersicht: III&nbsp;Referenzteil"></a></td><td class="navicon"><a href="../../manual_de.pdf" target="_parent"><img src="../icons/linkpdf.png" width="34" height="36" border="0" alt="PDF Version" title="PDF Version"></a></td><td class="navcopymod">Letzte &Auml;nderung: 6.9.2022<br>Copyright &copy; 1999-2022 Quality First Software GmbH</td>
</tr>
</table>
</body>
</html>
